### PR TITLE
Update the upgrade notes

### PR DESCRIPTION
**Pull request by richvdh**
* make it clear that if you installed from a package manager, you should use
   that to upgrade

 * Document the new way of getting the server version (cf #4878)

 * Write some words about downgrading.

### CODE REVIEW

File: UPGRADE.rst [hunk 1]
```diff
@@ -2,52 +2,76 @@ Upgrading Synapse
...
 
-   .. code:: bash
+     .. code:: bash
 
-       pip install --upgrade matrix-synapse[all]
+       source ~/synapse/env/bin/activate
 
-       # restart synapse
-       synctl restart
+  2. If synapse was installed using pip then upgrade to the latest version by
```
Comment (anoadragon453) [inline UPGRADE.rst]:
```suggestion
  2. If Synapse was installed using pip then upgrade to the latest version by
```

File: UPGRADE.rst [hunk 2]
```diff
@@ -2,52 +2,76 @@ Upgrading Synapse
...
+     running:
 
+     .. code:: bash
 
-   If synapse was installed using git then upgrade to the latest version by
-   running:
+       pip install --upgrade matrix-synapse
 
-   .. code:: bash
+     If synapse was installed using git then upgrade to the latest version by
```
Comment (anoadragon453) [inline UPGRADE.rst]:
```suggestion
     If Synapse was installed using git then upgrade to the latest version by
```

File: UPGRADE.rst [hunk 3]
```diff
@@ -2,52 +2,76 @@ Upgrading Synapse
...
+     running:
 
-       # Pull the latest version of the master branch.
+     .. code:: bash
        git pull
+       pip install --upgrade .
 
-       # Update synapse and its python dependencies.
-       pip install --upgrade .[all]
+   3. Restart synapse:
```
Comment (anoadragon453) [inline UPGRADE.rst]:
```suggestion
   3. Restart Synapse:
```

File: UPGRADE.rst [hunk 4]
```diff
@@ -2,52 +2,76 @@ Upgrading Synapse
...
+  for upgrading those packages.
 
-   .. code:: bash
+* If synapse was installed from source, then:
 
-       source ~/synapse/env/bin/activate
-
-2. If synapse was installed using pip then upgrade to the latest version by
-   running:
+  1. Activate the virtualenv before upgrading. For example, if synapse is
```
Comment (anoadragon453) [inline UPGRADE.rst]:
```suggestion
  1. Activate the virtualenv before upgrading. For example, if Synapse is
```

File: UPGRADE.rst [hunk 5]
```diff
@@ -2,52 +2,76 @@ Upgrading Synapse
...
-   If synapse was installed using git then upgrade to the latest version by
-   running:
+       pip install --upgrade matrix-synapse
 
-   .. code:: bash
+     If synapse was installed using git then upgrade to the latest version by
+     running:
 
-       # Pull the latest version of the master branch.
+     .. code:: bash
```
Comment (anoadragon453) [inline UPGRADE.rst]: I'm not sure what the correct RST syntax is here, but this bit is currently broken: https://github.com/matrix-org/synapse/blob/679193744d89756721ec16369647ae05cb4cbd31/UPGRADE.rst#upgrading-synapse

File: UPGRADE.rst [hunk 6]
```diff
@@ -2,52 +2,76 @@ Upgrading Synapse
...
+     source env/bin/activate
+     pip install matrix-synapse==<version>
+
+* Debian:
+
+  .. code:: bash
+
+     # replace `1.3.0` and `stretch` accordingly:
+     wget https://packages.matrix.org/debian/pool/main/m/matrix-synapse-py3/matrix-synapse-py3_1.3.0+stretch1_amd64.deb
+     dpkg -i matrix-synapse-py3_1.3.0+stretch1_amd64.deb
```
Comment (anoadragon453) [inline UPGRADE.rst]: Might be nice to add the git version. I assume it's just a:

```
git checkout release-<version>
pip install --upgrade .
```

File: UPGRADE.rst [hunk 7]
```diff
@@ -2,52 +2,76 @@ Upgrading Synapse
...
+
+In general, you will need to undo any changes made during the upgrade process,
+for example:
+
+* pip:
+
+  .. code:: bash
+
+     source env/bin/activate
+     pip install matrix-synapse==<version>
```
Comment (anoadragon453) [inline UPGRADE.rst]: Odd we have `<version>` here and "replace `1.3.0` below in the next section. Could we make this consistent please?

File: UPGRADE.rst [hunk 8]
```diff
@@ -2,52 +2,76 @@ Upgrading Synapse
...
 
-1. If synapse was installed in a virtualenv then activate that virtualenv before
-   upgrading. If synapse is installed in a virtualenv in ``~/synapse/env`` then
-   run:
+* If Synapse was installed using `prebuilt packages
+  <INSTALL.md#prebuilt-packages>`_, you will need to follow the normal process
+  for upgrading those packages.
 
-   .. code:: bash
+* If synapse was installed from source, then:
```
Comment (richvdh) [inline UPGRADE.rst]:
```suggestion
* If Synapse was installed from source, then:
```

File: UPGRADE.rst (hunk 6 above)
Comment (richvdh) [reply UPGRADE.rst]: I'm kinda assuming that if you've installed from git in the first place, you know how to go back to an earlier tag.

File: UPGRADE.rst (hunk 5 above)
Comment (richvdh) [reply UPGRADE.rst]: fixed.

File: UPGRADE.rst (hunk 7 above)
Comment (richvdh) [reply UPGRADE.rst]: ok, done.

**Pull request merged**